### PR TITLE
Docs: update run reference

### PR DESCRIPTION
**Pull request by dvdksn**
**- What I did**

Updated the `docker run` CLI reference page.

**- How I did it**

w/ input from @rumpl 

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by vvoland — APPROVED** [review]

**Review by sam-thibault — APPROVED** [review]

**Comment by dvdksn** [comment]
rdy for merge (I lack permissions)

File: docs/reference/commandline/run.md [hunk 1]
```diff
@@ -15,137 +15,106 @@ Aliases:
...
-      --init                          Run an init inside the container that forwards signals and reaps processes
-  -i, --interactive                   Keep STDIN open even if not attached
-      --io-maxbandwidth string        Maximum IO bandwidth limit for the system drive (Windows only)
-                                      (Windows only). The format is `<number><unit>`.
-                                      Unit is optional and can be `b` (bytes per second),
-                                      `k` (kilobytes per second), `m` (megabytes per second),
-                                      or `g` (gigabytes per second). If you omit the unit,
-                                      the system uses bytes per second.
-                                      --io-maxbandwidth and --io-maxiops are mutually exclusive options.
-      --io-maxiops uint               Maximum IOps limit for the system drive (Windows only)
```
Comment (thaJeztah) [inline docs/reference/commandline/run.md:80]: Looks like we lost the `--io-maxbandwidth` and `--io-maxiops` options (as they're only available on Windows)

File: docs/reference/commandline/run.md [hunk 2]
```diff
@@ -15,137 +15,106 @@ Aliases:
...
-      --memory-swappiness int         Tune container memory swappiness (0 to 100) (default -1)
-      --mount value                   Attach a filesystem mount to the container (default [])
-      --name string                   Assign a name to the container
-      --network-alias value           Add network-scoped alias for the container (default [])
-      --network string                Connect a container to a network
-                                      'bridge': create a network stack on the default Docker bridge
-                                      'none': no networking
-                                      'container:<name|id>': reuse another container's network stack
-                                      'host': use the Docker host network stack
-                                      '<network-name>|<network-id>': connect to a user-defined network
```
Comment (thaJeztah) [inline docs/reference/commandline/run.md:105]: Looks like there were some manually written additions here (describing the different network modes) that are not printed on the CLI itself. Those were removed now (should we make the cli itself print these?)

File: docs/reference/commandline/run.md [hunk 3]
```diff
@@ -15,137 +15,106 @@ Aliases:
...
-      --pid string                    PID namespace to use
-      --pids-limit int                Tune container pids limit (set -1 for unlimited)
-      --privileged                    Give extended privileges to this container
-  -p, --publish value                 Publish a container's port(s) to the host (default [])
-  -P, --publish-all                   Publish all exposed ports to random ports
-      --pull string                   Pull image before running ("always"|"missing"|"never") (default "missing")
-  -q, --quiet                         Suppress the pull output
-      --read-only                     Mount the container's root filesystem as read only
-      --restart string                Restart policy to apply when a container exits (default "no")
-                                      Possible values are : no, on-failure[:max-retry], always, unless-stopped
```
Comment (thaJeztah) [inline docs/reference/commandline/run.md:118]: Same here ("Possible values are...")

File: docs/reference/commandline/run.md [hunk 4]
```diff
@@ -15,137 +15,106 @@ Aliases:
...
-      --rm                            Automatically remove the container when it exits
-      --runtime string                Runtime to use for this container
-      --security-opt value            Security Options (default [])
-      --shm-size bytes                Size of /dev/shm
-                                      The format is `<number><unit>`. `number` must be greater than `0`.
-                                      Unit is optional and can be `b` (bytes), `k` (kilobytes), `m` (megabytes),
-                                      or `g` (gigabytes). If you omit the unit, the system uses bytes.
```
Comment (thaJeztah) [inline docs/reference/commandline/run.md:125]: And here ("the format is...")

File: docs/reference/commandline/run.md [hunk 5]
```diff
@@ -15,137 +15,106 @@ Aliases:
...
-      --stop-timeout int              Timeout (in seconds) to stop a container
-      --storage-opt value             Storage driver options for the container (default [])
-      --sysctl value                  Sysctl options (default map[])
-      --tmpfs value                   Mount a tmpfs directory (default [])
-  -t, --tty                           Allocate a pseudo-TTY
-      --ulimit value                  Ulimit options (default [])
-  -u, --user string                   Username or UID (format: <name|uid>[:<group|gid>])
-      --userns string                 User namespace to use
-                                      'host': Use the Docker host user namespace
-                                      '': Use the Docker daemon user namespace specified by `--userns-remap` option.
```
Comment (thaJeztah) [inline docs/reference/commandline/run.md:137]: Here as well (wonder if we had these in the past or if they were always docs-only)

File: docs/reference/commandline/run.md [hunk 6]
```diff
@@ -15,137 +15,106 @@ Aliases:
...
-      --uts string                    UTS namespace to use
-  -v, --volume value                  Bind mount a volume (default []). The format
-                                      is `[host-src:]container-dest[:<options>]`.
-                                      The comma-delimited `options` are [rw|ro],
-                                      [z|Z], [[r]shared|[r]slave|[r]private],
-                                      [delegated|cached|consistent], and
-                                      [nocopy]. The 'host-src' is an absolute path
-                                      or a name value.
```
Comment (thaJeztah) [inline docs/reference/commandline/run.md:145]: Same here

File: docs/reference/commandline/run.md [hunk 7]
```diff
@@ -582,12 +550,14 @@ still store what's been written to `STDERR` and `STDOUT`.
...
 ```
 
-This is how piping a file into a container could be done for a build.
+This is a way of using `--attach` to pipe a build file into a container.
 The container's ID will be printed after the build is done and the build
 logs could be retrieved using `docker logs`. This is
 useful if you need to pipe a file or something else into a container and
 retrieve the container's ID once the container has finished running.
 
+See also [the `docker cp` command](/engine/reference/commandline/cp/).
```
Comment (thaJeztah) [inline docs/reference/commandline/run.md:559]: We may need to either use a full (`docs.docker.com/....`) URL for this if we want the link to work on GitHub, or a relative URL (`cp.md`)

**Pull request merged**